.github/workflows/docker-publish.yml

```diff
@@ -17,6 +17,11 @@ on:
 jobs:
   docker:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
```

Comment:
Afaict, this workflow is probably more or less broken because it _runs_ _on_ `pull_request` but  the permission requests you're making won't work for pull_requests coming from forks when the creators don't have write access to the destination repository.

You have an `if` later in a step, but depending on what you're trying to achieve, it should either be moved from the step to the job, or reproduced in additional steps.